oidc-service-impl/src/main/java/io/mosip/esignet/services/ConsentHelperService.java

```diff
@@ -0,0 +1,224 @@
+            consentRequest.setAuthorizationScopes(authorizeScopes);
+            consentService.saveUserConsent(consentRequest);
+        }
+    }
+
+    private void setIsAcceptedBasedOnClaims(List<String> acceptedClaims, Claims requestedClaims) {
+        Map<String ,ClaimDetail> userInfo = requestedClaims.getUserinfo();
+        for (String acceptedClaim : acceptedClaims) {
+            if (userInfo.containsKey(acceptedClaim)) {
+                ClaimDetail claimDetail = userInfo.get(acceptedClaim);
+                if(claimDetail!= null){
+                    claimDetail.setAccepted(true);
+                } else {
+                    claimDetail = new ClaimDetail();
+                    claimDetail.setAccepted(true);
+                }
+                userInfo.put(acceptedClaim, claimDetail);
+            }
+        }
+    }
+
+
+    private static ConsentAction validateConsentAction(OIDCTransaction transaction, Consent consent) {
+        if(consent == null) {
+            return ConsentAction.CAPTURE;
+        }
+        //validate requested claims
+        Claims requestedClaims = transaction.getRequestedClaims();
+        List<String> requestedScopes = transaction.getRequestedAuthorizeScopes();
+
+        if(((requestedClaims == null ||
+                (requestedClaims.getId_token().isEmpty() && requestedClaims.getUserinfo().isEmpty()))
+                && requestedScopes.isEmpty())
+        ) {
+            return ConsentAction.NOCAPTURE;
+        }
+
+        //validate consented claims
+        if(requestedClaims!= null && consent.getClaims() != null &&!requestedClaims.isEqualToIgnoringAccepted(consent.getClaims())){
+            return ConsentAction.CAPTURE;
+        }
+
+        //validate consented scopes
+        Set<String> acceptedScopes = consent.getAuthorizationScopes()
+                .entrySet()
+                .stream().filter(Map.Entry::getValue)
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toSet());
+        if(!requestedScopes.isEmpty()
+                && ( !new HashSet<>(requestedScopes).containsAll(consent.getAuthorizationScopes().keySet()) ||
+                !new HashSet<>(requestedScopes).containsAll(acceptedScopes)
+        )){
+            return ConsentAction.CAPTURE;
+        }
+        return ConsentAction.NOCAPTURE;
+    }
+
+    private boolean validateSignature(OIDCTransaction transaction, Consent consent){
+        return true;
+
+    }
+    private String generateSignedObject(OIDCTransaction transaction, Consent consent){
+        List<String> acceptedClaims = transaction.getAcceptedClaims();
+        List<String> permittedScopes = transaction.getPermittedScopes();
+        Collections.sort(acceptedClaims);
+        Collections.sort(acceptedClaims);
+        JWTClaimsSet claimsSet = new JWTClaimsSet.Builder()
+                .claim("acceptedClaims", acceptedClaims)
+                .claim("authorizeScopes", permittedScopes)
```

Comment:
we should also have timestamps

Reply:
this is still wip, signature will be taken in next sprint, will remove the function from the PR